### PR TITLE
Fixes redirect on unauthorised error

### PR DESCRIPTION
**Pull request by insacuri**

### CODE REVIEW

File: response_operations_ui/views/sign_in.py [hunk 1]
```diff
@@ -28,8 +28,9 @@ def sign_in():
         if 'token' in response_json:
             user = User(response_json['token'])
             login_user(user)
-
-            next_url = request.args.get('next')
-            return redirect(next_url or url_for('home_bp.home'))
+            if 'next' in session:
+                return redirect(session['next'])
```
Comment (andrewmil) [inline response_operations_ui/views/sign_in.py]: This case isn't tested

File: response_operations_ui/views/sign_in.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import logging
 
-from flask import Blueprint, redirect, render_template, request, url_for
+from flask import Blueprint, redirect, render_template, request, url_for, session
```
Comment (JamesGardiner) [inline response_operations_ui/views/sign_in.py]: I'm going to be nit picking but this should be alphabetized

Comment (insacuri) [reply response_operations_ui/views/sign_in.py]: Fixed

File: response_operations_ui/views/sign_in.py (hunk 1 above)
Comment (insacuri) [reply response_operations_ui/views/sign_in.py]: Tested now

File: response_operations_ui/views/sign_in.py [hunk 3]
```diff
@@ -31,8 +31,9 @@ def sign_in():
         if 'token' in response_json:
             user = User(response_json['token'])
             login_user(user)
-
-            next_url = request.args.get('next')
-            return redirect(next_url or url_for('home_bp.home'))
+            if 'next' in session:
+                return redirect(session['next'])
+            else:
```
Comment (jcox-dev) [inline response_operations_ui/views/sign_in.py]: 🎨 superfluous `else` statement here.

**Pull request merged**